src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/DesktopOsHelper.cs

```diff
@@ -55,6 +55,16 @@ public static bool IsMac()
 #endif
         }
 
+        public static bool IsDarwin()
+        {
+#if MAC
```

Comment:
This flag isn't in use!

MSAL today can run on MacOS only on `.NET` and `.NETSTANDARD` target frameworks. MSAL doesn't explicitly target `net8-maccatalyst`, so if an app targets net8-maccatalyst it will end up hitting MSAL `net8` version I believe. 

IMO you only need to take care of `net8` support as full `maccatalyst` MSAL support is tracked seperately (lots on interest on that) but is currently blocked due to caching concerns.